web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -104,15 +108,21 @@ export function Resources() {
 
   useEffect(() => {
     getPinnedResources();
+    setSelectedResources([]);
+    setUpdatePinnedResources(makeEmptyAttempt());
   }, [clusterId, getPinnedResources]);
 
   const pinnedResources = getPinnedResourcesAttempt.data || [];
 
-  const [updatePinnedResourcesAttempt, updatePinnedResources] = useAsync(
+  const [
+    updatePinnedResourcesAttempt,
+    updatePinnedResources,
+    setUpdatePinnedResources,
+  ] = useAsync(
     useCallback(
```

Comment:
We removed `useCallback` from `updateClusterPinnedResources`, we can remove it here as well.